src/main/java/com/amazon/opendistroforelasticsearch/sql/query/ESActionFactory.java

```diff
@@ -91,6 +91,13 @@ else if(isJoin(sqlExpr,sql)){
                 else {
                     sqlExpr.accept(new TermFieldRewriter(client));
                     Select select = new SqlParser().parseSelect(sqlExpr);
+                    if (select.isSelectAll() && !select.getFields().isEmpty()) {
+                        for (int i = 0; i < select.getFields().size(); ++i) {
+                            if (select.getFields().get(i) != null && !select.getFields().get(i).isNested()) {
+                                throw new SqlParseException("Other expressions may not be present in the select list when '*' is used");
```

Comment:
**Nested** expressions may not be present, and you could add the expression itself (field